store/dataset.js

```diff
@@ -135,6 +135,39 @@ function guessColumnType(data) {
 }
 
 export const actions = {
+  async setCsvFiles({ commit, state, dispatch }, files) {
+    commit('setCsvFiles', files)
+    await dispatch(
+      'setVegaSpecData',
+      {
+        type: 'csv',
+        index: state.csvIndex,
+      },
+      { root: true }
+    )
+  },
+  async setGeojsonFiles({ commit, state, dispatch }, files) {
+    commit('setGeojsonFiles', files)
+    await dispatch(
+      'setVegaSpecData',
+      {
+        type: 'geojson',
+        index: state.geoIndex,
+      },
+      { root: true }
+    )
+  },
+  async setTopojsonFiles({ commit, state, dispatch }, files) {
+    commit('setTopojsonFiles', files)
+    await dispatch(
+      'setVegaSpecData',
+      {
+        type: 'topojson',
+        index: state.geoIndex,
```

Comment:
Is "geoIndex" the right one here?!

Reply:
Yep, so the only reason theres a `csvIndex` and a `geoIndex` is because you can (though its disabled right now) combine a csv and a geojson/topojson file. The `csvIndex` is always for the csv file dropdown the `geoIndex` is used for both the geojson and topojson file dropdown because they never appear at the same time